mlflow/store/azure_blob_artifact_repo.py

```diff
@@ -0,0 +1,123 @@
+import os
+import re
+
+from six.moves import urllib
+
+from mlflow.entities.file_info import FileInfo
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.utils.file_utils import build_path, get_relative_path, TempDir
+
+
+class AzureBlobArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on Azure Blob Storage.
+
+    This repository is used with URIs of the form
+    ``wasbs://<container-name>@<ystorage-account-name>.blob.core.windows.net/<path>``,
+    following the same URI scheme as Hadoop on Azure blob storage. It requires that your Azure
+    storage access key be available in the environment variable ``AZURE_STORAGE_ACCESS_KEY``.
+    """
+
+    def __init__(self, artifact_uri, client=None):
+        if client:
+            self.client = client
+        else:
+            from azure.storage.blob import BlockBlobService
+            (_, account, _) = AzureBlobArtifactRepository.parse_wasbs_uri(artifact_uri)
+            if "AZURE_STORAGE_CONNECTION_STRING" in os.environ:
+                self.client = BlockBlobService(
+                    account_name=account,
+                    connection_string=os.environ.get("AZURE_STORAGE_CONNECTION_STRING"))
+            elif "AZURE_STORAGE_ACCESS_KEY" in os.environ:
+                self.client = BlockBlobService(
+                    account_name=account,
+                    account_key=os.environ.get("AZURE_STORAGE_ACCESS_KEY"))
+            else:
+                raise Exception("You need to set one of AZURE_STORAGE_CONNECTION_STRING or "
+                                "AZURE_STORAGE_ACCESS_KEY to access Azure storage.")
+        super(AzureBlobArtifactRepository, self).__init__(artifact_uri)
```

Comment:
nit: maybe we can call super first, and then in the `if client:` condition, return early. Looks more normal (super first) and avoids a level of indentation.